x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -190,11 +190,41 @@ export function trainedModelsRoutes(
             // we don't need to fill kibana's log with these messages.
             mlLog.debug(e);
           }
+          const enabledFeatures = getEnabledFeatures();
+          try {
+            if (enabledFeatures.dfa) {
+              const jobIds = result.map((model) => {
+                let id = model.metadata?.analytics_config?.id;
+                if (id) {
+                  id = `${id}*`;
```

Comment:
I don't think we should be modifying the IDs. This could lead to unexpected behaviour. e.g. if we have two jobs `foo` and `foo2`. wildcarding `foo*` will match both.

In my previous suggestion I overlooked the fact that comma separating the IDs to pass to `getDataFrameAnalytics` will throw a 404 if any of them are missing.

I still don't think we should call `getDataFrameAnalytics` in a loop and so I think the simplest solution would be to fetch all DFA jobs and just loop over them looking for the IDs we want.
That call isn't expensive. We could also be clever and only use the job ID if there is only one model in the results. As it's likely this endpoint will be called with either one model ID or no model IDs.

I had a go at writing up what I'm thinking. I got carried away trying to make it performant.

```ts
const filteredModels = filterForEnabledFeatureModels<TrainedModelConfigResponse>(
  result,
  getEnabledFeatures()
);
const dfaJobIdMap = filteredModels.reduce<Record<string, string>>((c, m) => {
  const id = m.metadata?.analytics_config?.id;
  if (id !== undefined) {
    c[m.model_id] = id;
  }
  return c;
}, {});

const jobIds = Object.values(dfaJobIdMap);

if (jobIds.length === 0) {
  // return early, there are no dfa jobs
  return response.ok({
    body: filteredModels,
  });
}

let dfaJobs: estypes.MlDataframeAnalyticsSummary[] = [];
try {
  const jobs =
    jobIds.length === 1
      ? await mlClient.getDataFrameAnalytics({
          id: jobIds[0],
        })
      : await mlClient.getDataFrameAnalytics();

  dfaJobs = jobs.data_frame_analytics;
} catch (e) {
  //
}

for (const model of filteredModels) {
  const dfaJob = dfaJobs.find((j) => j.id === dfaJobIdMap[model.model_id]);
  model.origin_job_exists = dfaJob !== undefined;
}

return response.ok({
  body: filteredModels,
});
```

Reply:
We've discussed this offline and can't think of a situation where adding `*` to each job might cause the wrong job to be matched.
I'm still not a fan of adding these `*` characters to work around the fact the es endpoint will throw a 404 if one job can't be found. But there's not a compelling reason to demand this change.